xcc/settings.py

```diff
@@ -87,15 +87,45 @@ class Config:  # pylint: disable=missing-class-docstring
         env_file = get_path_to_env_file()
         env_prefix = get_name_of_env_var()
 
+    def _sanity_check(self, key: str, val: str) -> None:
+        """
+        Check for conditions that make saving the env_file
+        dangerous to the user.
+
+        Parameters
+        ----------
+        key : str
+            env file key
+        val : str
+            env file value
+
+        Raises
+        ------
+        Exception
+            When the value should not be saved to the env_file.
+
+        Returns
+        -------
+        None
+            Has no effect if all is safe.
+
+        """
+        # Unprintables in REFRESH_TOKEN assumed never to be valid (0x0a, 0x20, etc.)
+
+        if key == "REFRESH_TOKEN" and val is not None and not val.isprintable():
+            raise Exception("REFRESH_TOKEN contains non-printable character(s)")
+
     def save(self) -> None:
         """Saves the current settings to the .env file."""
+
         env_file = Settings.Config.env_file
         env_dir = os.path.dirname(env_file)
         os.makedirs(env_dir, exist_ok=True)
 
         saved = dotenv_values(dotenv_path=env_file)
 
         for key, val in self.dict().items():
+            self._sanity_check(key, val)  # check for obviously bad settings
```

Comment:
I would argue this comment does not add any information which is not already disclosed by the function name:
```suggestion
            self._sanity_check(key, val)
```